internal/k6runner/k6runner.go

```diff
@@ -26,20 +26,33 @@ type Runner interface {
 	Run(ctx context.Context, script []byte) (*RunResponse, error)
 }
 
-func New(uri string) Runner {
+type RunnerOpts struct {
+	Uri           string
+	BlacklistedIP string
+}
+
+func New(opts RunnerOpts) Runner {
 	var r Runner
 	logger := zerolog.Nop()
 
-	if strings.HasPrefix(uri, "http") {
+	if strings.HasPrefix(opts.Uri, "http") {
 		r = &HttpRunner{
-			url:    uri,
+			url:    opts.Uri,
 			logger: &logger,
 		}
 	} else {
+		var blacklistIP string
+		if opts.BlacklistedIP != "" {
+			blacklistIP = opts.BlacklistedIP
+		} else {
+			blacklistIP = "10.0.0.0/8"
+		}
+
 		r = &LocalRunner{
-			k6path: uri,
-			logger: &logger,
-			fs:     afero.NewOsFs(),
+			k6path:        opts.Uri,
+			logger:        &logger,
+			fs:            afero.NewOsFs(),
+			blacklistedIP: blacklistIP,
 		}
```

Comment:
I feel this is going to start getting out of hand as we add support for the other TODO items in the way we call the local runner.

What do you think about:

```
r := LocalRunner{
    k6path: "k6",
    logger: &logger,
    fs: afero.NewOsFs(),
    blacklistedIP: blacklistIP,
}

r.withOpts(opts)
```

and then withOpts is something like:

```
func (r *LocalRunner) withOpts(opts RunnerOpts) {
    if opts.blacklistedIPs != "" {
        r.blacklistedIPs = opts.blacklistedIPs
    }
    // ...
}
```

in that way you have a list of default values, and you can override those with options.

The other alternative is that you go with functional options, which I don't like too much to be honest, but it's a common pattern.

Reply:
I like this suggestion! I'm also not too a huge fan of functional options so this works for me